pangeo_forge_recipes/recipes/xarray_zarr.py

```diff
@@ -137,163 +137,109 @@ def chunk_position(chunk_key: ChunkKey) -> int:
     return subset_factor * concat_idx + subset_idx
 
 
-def cache_input_metadata(
-    input_key: InputKey,
-    metadata_cache: Optional[MetadataTarget],
-    file_pattern: FilePattern,
-    input_cache: Optional[CacheFSSpecTarget],
-    cache_inputs: bool,
-    copy_input_to_local_file: bool,
-    xarray_open_kwargs: dict,
-    delete_input_encoding: bool,
-    process_input: Optional[Callable[[xr.Dataset, str], xr.Dataset]],
-) -> None:
-    if metadata_cache is None:
-        raise ValueError("metadata_cache is not set.")
-    logger.info(f"Caching metadata for input '{input_key!s}'")
-    with open_input(
-        input_key,
-        file_pattern=file_pattern,
-        input_cache=input_cache,
-        cache_inputs=cache_inputs,
-        copy_input_to_local_file=copy_input_to_local_file,
-        xarray_open_kwargs=xarray_open_kwargs,
-        delete_input_encoding=delete_input_encoding,
-        process_input=process_input,
-    ) as ds:
-        input_metadata = ds.to_dict(data=False)
-        metadata_cache[_input_metadata_fname(input_key)] = input_metadata
-
-
-def cache_input(
-    input_key: InputKey,
-    cache_inputs: bool,
-    input_cache: Optional[CacheFSSpecTarget],
-    file_pattern: FilePattern,
-    cache_metadata: bool,
-    copy_input_to_local_file: bool,
-    xarray_open_kwargs: dict,
-    delete_input_encoding: bool,
-    process_input: Optional[Callable[[xr.Dataset, str], xr.Dataset]],
-    metadata_cache: Optional[MetadataTarget],
-) -> None:
-    if cache_inputs:
-        if file_pattern.is_opendap:
+def cache_input(config: XarrayZarrRecipeConfig, input_key: InputKey) -> None:
```

Comment:
Here's an optional style suggestion: 

There's a python feature that I enjoy that seems to be a good fit for these functions ([keyword only arguments](https://www.asmeurer.com/python3-presentation/slides.html#8)):
```
def cache_input(input_key: InputKey, *, config: XarrayZarrRecipeConfig) -> None:
...
```
In order to invoke this function, the caller would have to write:
```
cache_input(my_input_key, config=config)
```


I think this structure is a good fit for these functions since: 
- it seems more natural for the first argument to imply the value being operated on, instead of a shared context
- it forces the caller to pass in the config as a key-word argument. I think this is less confusing to callers. It helps people remember to include state; error messages help point he developer in the right direction.
- it prevents another object from accidentally being passed in as the config.